CMakeLists.txt

```
@@ -50,6 +50,10 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
+# setting some extra variables needed by external projects
+set(ROOT_GLOBAL_SOURCE_DIR "${CMAKE_SOURCE_DIR}")
+set(ROOT_GLOBAL_BINARY_DIR "${CMAKE_BINARY_DIR}")
```

```suggestion
set(ROOT_MAIN_SOURCE_DIR "${CMAKE_SOURCE_DIR}")
set(ROOT_MAIN_BINARY_DIR "${CMAKE_BINARY_DIR}")
```

I think using main reads better.

---

fixed